yapapi/rest/activity.py

```diff
@@ -138,6 +139,28 @@ def __init__(self, command: str, message: Optional[str] = None, stderr: Optional
         self.stderr = stderr
 
     def __str__(self) -> str:
+        msg = self._generic_msg()
+        if self.message is not None and "State error: Invalid state: StatePair" in self.message:
+            msg = "\n".join((msg, self._invalid_state_msg()))
+        return msg
+
+    def _invalid_state_msg(self) -> str:
+        assert isinstance(self.message, str)
+        match = re.match(r".*StatePair\((\w+), (\w+)\)$", self.message)
+        assert match is not None
+        states = match.groups()
+        assert len(states) == 2
+
+        if states == ("Initialized", "None"):
+            return "Activity is initialized - deploy() command is expected now"
+        if states == ("Deployed", "None"):
+            return "Activity is deployed - start() command is expected now"
+        if states == ("Ready", "None"):
+            command = list(self.command.keys())[0]  # type: ignore # https://github.com/golemfactory/yapapi/issues/701
+            return f"There was already a succesful start() - sending {command}() is not allowed"
+        return f"This command is not allowed when activity is in state {states[0]}"
```

Comment:
Will it be obvious to the user what command we're referring to here (i.e. by saying "this command")?

Reply:
1. I'm don't know any scenario that leads to the last message (this non-if), I can only recreate first 3 messages. Maybe it's not possible at all?
2. Full error would look like this:
```
[2021-10-01T17:39:52.076+0200 WARNING yapapi.summary] [Job 1] Activity failed on provider 'mbenke'; reason: Command '{'deploy': {}}' failed on provider; message: 'Local service error: State error: Invalid state: StatePair(SomeState, None)'
This command is not allowed when activity is in state SomeState
```
So I think it is clear enough (command deploy failed...).

Reply:
I could change it to `command` (the one from line 159), but I'm not sure what this "command" is supposed to be (#701) so I wanted to make no assumptions (e.g. in case if it was a `str`).

Reply:
hmm.... I'm generally unsure about this PR ... the sequence of commands and generally what can and what cannot be done when is determined and completely dependent on a specific runtime ... I'm not happy about adding logic to yapapi that's dependent on behavior of a specific runtime ...

Reply:
It would be better to add a better message here:
https://github.com/golemfactory/yagna/blob/c4205cda0336a67142534be0043f010163237be2/exe-unit/src/state.rs#L32
@mfranciszkiewicz

Reply:
This is my PR to `yagna` that changes the error message:
https://github.com/golemfactory/yagna/pull/1631/files

Reply:
This PR was already merged to `release/v0.8`.
https://github.com/golemfactory/yagna/pull/1632